{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "86c57265",
   "metadata": {},
   "source": [
    "## Walkthrough for annotating a mutation list with database data using annovar and clinscore\n",
    "\n",
    "#### Oftentimes you receive a mutation list of any sort and you need to annotate it with database info to have an idea about the position and relevance of the data. More than often, this task can be subdivided into several or all of the following steps:\n",
    "+ wrangle the data into the format (ID) Chr Start End Ref Alt .....\n",
    "+ if coords are on hg19, convert it to hg38 to use this annotation set (same procedure can be applied if converting back to hg19 is required+ annotate with annovar on command line using any of a set of databases\n",
    "+ apply clinscore calculation to get relevance of mutations in a clear (scalar) fashion"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8b801f8",
   "metadata": {},
   "source": [
    "#### init paths and code base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bed10000",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:42:57.537342Z",
     "start_time": "2022-02-06T11:42:57.522864Z"
    }
   },
   "outputs": [],
   "source": [
    "# some sensible settings for better output\n",
    "import os\n",
    "import pandas as pd\n",
    "from IPython.display import display\n",
    "pd.set_option('display.max_columns', None)\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\"\n",
    "pd.set_option('max_colwidth', 200)\n",
    "\n",
    "\n",
    "# get the code\n",
    "import sys\n",
    "sys.path.append('../code')\n",
    "from script_utils import show_output\n",
    "\n",
    "# paths from environ file\n",
    "home = os.environ['HOME']\n",
    "# you need static files for the annovar annotation\n",
    "static_path = os.path.join(os.environ['STATIC'], \"annotation/clinical\")\n",
    "# here, set the working directory where your files are saved\n",
    "\n",
    "workdir = os.path.join(\"../\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90f7e693",
   "metadata": {},
   "source": [
    "### load mutation file and inspect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f306abe7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:42:58.151419Z",
     "start_time": "2022-02-06T11:42:58.105625Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Location</th>\n",
       "      <th>Pos.</th>\n",
       "      <th>Type</th>\n",
       "      <th>Nuc Change</th>\n",
       "      <th>check Nuc</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>AA Change</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Hint</th>\n",
       "      <th>web Ref.</th>\n",
       "      <th>c. HGVS</th>\n",
       "      <th>p. HGVS</th>\n",
       "      <th>mut Entry</th>\n",
       "      <th>mut Effect</th>\n",
       "      <th>TValidation</th>\n",
       "      <th>MValidation</th>\n",
       "      <th>Weighting</th>\n",
       "      <th>MutDB:Classification</th>\n",
       "      <th>MutDB:Has Weblinks</th>\n",
       "      <th>ClinVitae:Classification</th>\n",
       "      <th>ClinVitae:Reported Classification</th>\n",
       "      <th>COSMIC:Pubmed_PMID</th>\n",
       "      <th>ClinVar:Clinical Significance</th>\n",
       "      <th>ClinVar:ReviewStatus</th>\n",
       "      <th>1000Genomes:AF</th>\n",
       "      <th>1000Genomes:AVGPOST</th>\n",
       "      <th>1000Genomes:EUR_AF</th>\n",
       "      <th>dbSNP:ASS</th>\n",
       "      <th>dbSNP:COMMON</th>\n",
       "      <th>dbSNP:DSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E16</td>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>60.0</td>\n",
       "      <td>60% (4570)   [60% (2481) / 59% (2089)]</td>\n",
       "      <td>H -&gt; H (706)</td>\n",
       "      <td>synonymous, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>rs151207425 (1000Genomes; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.2118T&gt;C</td>\n",
       "      <td>p.His706</td>\n",
       "      <td>0/7 of 2382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E16</td>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58% (3786)   [58% (2074) / 58% (1712)]</td>\n",
       "      <td>H -&gt; H (706)</td>\n",
       "      <td>synonymous, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>rs151207425 (1000Genomes; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.2118T&gt;C</td>\n",
       "      <td>p.His706</td>\n",
       "      <td>0/7 of 2382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E9</td>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "      <td>OK</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99% (4939)   [99% (1759) / 100% (3180)]</td>\n",
       "      <td>G -&gt; R (388)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COSM1567768 (COSMIC), rs351855 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.1162G&gt;A</td>\n",
       "      <td>p.Gly388Arg</td>\n",
       "      <td>225/1122 of 2861</td>\n",
       "      <td>Polymorphismus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.299521</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2942</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E9</td>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "      <td>OK</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99% (4177)   [99% (1491) / 100% (2686)]</td>\n",
       "      <td>G -&gt; R (388)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COSM1567768 (COSMIC), rs351855 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.1162G&gt;A</td>\n",
       "      <td>p.Gly388Arg</td>\n",
       "      <td>225/1122 of 2861</td>\n",
       "      <td>Polymorphismus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.299521</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2942</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E12</td>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>60.0</td>\n",
       "      <td>60% (4196)   [61% (1860) / 59% (2336)]</td>\n",
       "      <td>E -&gt; D (522)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1566G&gt;T</td>\n",
       "      <td>p.Glu522Asp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NM_213647</td>\n",
       "      <td>E12</td>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>38.0</td>\n",
       "      <td>38% (2061)   [38% (817) / 38% (1244)]</td>\n",
       "      <td>E -&gt; D (522)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1566G&gt;T</td>\n",
       "      <td>p.Glu522Asp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>NM_005343</td>\n",
       "      <td>E2</td>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>C -&gt; T (het)</td>\n",
       "      <td>C=G</td>\n",
       "      <td>4.8</td>\n",
       "      <td>4.8% (349)   [6% (159) / 4.1% (190)]</td>\n",
       "      <td>5' UTR</td>\n",
       "      <td>5' UTR, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs41294870 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.-10C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/43 of 697 [c.-10C&gt;T (NM_001130442)]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>benign</td>\n",
       "      <td>no</td>\n",
       "      <td>Benign,Not Provided</td>\n",
       "      <td>Benign,not provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.035743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4.0</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>NM_005343</td>\n",
       "      <td>E2</td>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>C=G</td>\n",
       "      <td>29.0</td>\n",
       "      <td>29% (2459)   [33% (1122) / 27% (1337)]</td>\n",
       "      <td>5' UTR</td>\n",
       "      <td>5' UTR, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs41294870 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.-10C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/43 of 697 [c.-10C&gt;T (NM_001130442)]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>benign</td>\n",
       "      <td>no</td>\n",
       "      <td>Benign,Not Provided</td>\n",
       "      <td>Benign,not provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.035743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3.0</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>NM_203500</td>\n",
       "      <td>E3</td>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>C -&gt; A (het)</td>\n",
       "      <td>C=G</td>\n",
       "      <td>95.0</td>\n",
       "      <td>95% (9632)   [95% (4890) / 96% (4742)]</td>\n",
       "      <td>S -&gt; [STOP] (338)</td>\n",
       "      <td>stop gained, missense, protein altering, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1013C&gt;A</td>\n",
       "      <td>p.Ser338Ter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3.0</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>NM_203500</td>\n",
       "      <td>E3</td>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>C=G</td>\n",
       "      <td>50.0</td>\n",
       "      <td>50% (5186)   [49% (2692) / 51% (2494)]</td>\n",
       "      <td>S -&gt; [STOP] (338)</td>\n",
       "      <td>stop gained, missense, protein altering, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1013C&gt;A</td>\n",
       "      <td>p.Ser338Ter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>6.0</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>NM_002529</td>\n",
       "      <td>E17</td>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55% (3323)   [53% (1019) / 56% (2304)]</td>\n",
       "      <td>R -&gt; Q (780)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>rs35669708 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.2339G&gt;A</td>\n",
       "      <td>p.Arg780Gln</td>\n",
       "      <td>0/15 of 1598</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Conflicting interpretations of pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003794</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>7.0</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>NM_002529</td>\n",
       "      <td>E17</td>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>60.0</td>\n",
       "      <td>60% (3707)   [59% (1126) / 60% (2581)]</td>\n",
       "      <td>R -&gt; Q (780)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>primer position</td>\n",
       "      <td>rs35669708 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)</td>\n",
       "      <td>c.2339G&gt;A</td>\n",
       "      <td>p.Arg780Gln</td>\n",
       "      <td>0/15 of 1598</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Conflicting interpretations of pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003794</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>NM_020975</td>\n",
       "      <td>E13</td>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>47.0</td>\n",
       "      <td>47% (3315)   [49% (1801) / 45% (1514)]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2285-38G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1.0</td>\n",
       "      <td>RET</td>\n",
       "      <td>NM_020975</td>\n",
       "      <td>E13</td>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>OK</td>\n",
       "      <td>31.0</td>\n",
       "      <td>31% (2217)   [32% (1200) / 30% (1017)]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2285-38G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>7.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>E5</td>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>A -&gt; G (het)</td>\n",
       "      <td>A=T</td>\n",
       "      <td>94.0</td>\n",
       "      <td>94% (9509)   [94% (5096) / 95% (4413)]</td>\n",
       "      <td>K -&gt; E (132)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs747342068 (ClinVar; dbSNP; gnomAD), COSM1649370 (COSMIC)</td>\n",
       "      <td>c.394A&gt;G</td>\n",
       "      <td>p.Lys132Glu</td>\n",
       "      <td>0/7 of 6835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>no</td>\n",
       "      <td>Variant of uncertain significance, likely pathogenic,Variant of uncertain significance</td>\n",
       "      <td>Likely pathogenic,Uncertain significance</td>\n",
       "      <td>1.041699e+119</td>\n",
       "      <td>Conflicting interpretations of pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>8.0</td>\n",
       "      <td>TP53</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>E5</td>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>C</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>A=T</td>\n",
       "      <td>45.0</td>\n",
       "      <td>45% (5485)   [45% (3137) / 45% (2348)]</td>\n",
       "      <td>K -&gt; E (132)</td>\n",
       "      <td>missense, protein altering, coding sequence, exon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs747342068 (ClinVar; dbSNP; gnomAD), COSM1649370 (COSMIC)</td>\n",
       "      <td>c.394A&gt;G</td>\n",
       "      <td>p.Lys132Glu</td>\n",
       "      <td>0/7 of 6835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>distinct</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>no</td>\n",
       "      <td>Variant of uncertain significance, likely pathogenic,Variant of uncertain significance</td>\n",
       "      <td>Likely pathogenic,Uncertain significance</td>\n",
       "      <td>1.041699e+119</td>\n",
       "      <td>Conflicting interpretations of pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Index   Gene Transcript Location                                    Pos.  \\\n",
       "0     4.0  FGFR4  NM_213647      E16  103 (2118)   [chr5:g.176523707 (hg19)]   \n",
       "1     5.0  FGFR4  NM_213647      E16  103 (2118)   [chr5:g.176523707 (hg19)]   \n",
       "2     5.0  FGFR4  NM_213647       E9  105 (1162)   [chr5:g.176520243 (hg19)]   \n",
       "3     6.0  FGFR4  NM_213647       E9  105 (1162)   [chr5:g.176520243 (hg19)]   \n",
       "4     2.0  FGFR4  NM_213647      E12   47 (1566)   [chr5:g.176522377 (hg19)]   \n",
       "5     2.0  FGFR4  NM_213647      E12   47 (1566)   [chr5:g.176522377 (hg19)]   \n",
       "6     NaN   HRAS  NM_005343       E2            44   [chr11:g.534332 (hg19)]   \n",
       "7     4.0   HRAS  NM_005343       E2            44   [chr11:g.534332 (hg19)]   \n",
       "8     3.0  KEAP1  NM_203500       E3  374 (1013)   [chr19:g.10602565 (hg19)]   \n",
       "9     3.0  KEAP1  NM_203500       E3  374 (1013)   [chr19:g.10602565 (hg19)]   \n",
       "10    6.0  NTRK1  NM_002529      E17  134 (2339)   [chr1:g.156851382 (hg19)]   \n",
       "11    7.0  NTRK1  NM_002529      E17  134 (2339)   [chr1:g.156851382 (hg19)]   \n",
       "12    1.0    RET  NM_020975      E13         -38   [chr10:g.43613783 (hg19)]   \n",
       "13    1.0    RET  NM_020975      E13         -38   [chr10:g.43613783 (hg19)]   \n",
       "14    7.0   TP53  NM_000546       E5     19 (394)   [chr17:g.7578536 (hg19)]   \n",
       "15    8.0   TP53  NM_000546       E5     19 (394)   [chr17:g.7578536 (hg19)]   \n",
       "16    NaN    NaN        NaN      NaN                                     NaN   \n",
       "17    NaN      A        NaN      NaN                                     NaN   \n",
       "18    NaN      B        NaN      NaN                                     NaN   \n",
       "\n",
       "   Type     Nuc Change check Nuc  Unnamed: 8  \\\n",
       "0     C   T -> C (het)        OK        60.0   \n",
       "1     C   T -> C (het)        OK        58.0   \n",
       "2     C  G -> A (homo)        OK        99.0   \n",
       "3     C  G -> A (homo)        OK        99.0   \n",
       "4     C   G -> T (het)        OK        60.0   \n",
       "5     C   G -> T (het)        OK        38.0   \n",
       "6     C   C -> T (het)       C=G         4.8   \n",
       "7     C   G -> A (het)       C=G        29.0   \n",
       "8     C   C -> A (het)       C=G        95.0   \n",
       "9     C   G -> T (het)       C=G        50.0   \n",
       "10    C   G -> A (het)        OK        55.0   \n",
       "11    C   G -> A (het)        OK        60.0   \n",
       "12    C   G -> T (het)        OK        47.0   \n",
       "13    C   G -> T (het)        OK        31.0   \n",
       "14    C   A -> G (het)       A=T        94.0   \n",
       "15    C   T -> C (het)       A=T        45.0   \n",
       "16  NaN            NaN       NaN         NaN   \n",
       "17  NaN            NaN       NaN         NaN   \n",
       "18  NaN            NaN       NaN         NaN   \n",
       "\n",
       "                                   Coverage          AA Change  \\\n",
       "0    60% (4570)   [60% (2481) / 59% (2089)]       H -> H (706)   \n",
       "1    58% (3786)   [58% (2074) / 58% (1712)]       H -> H (706)   \n",
       "2   99% (4939)   [99% (1759) / 100% (3180)]       G -> R (388)   \n",
       "3   99% (4177)   [99% (1491) / 100% (2686)]       G -> R (388)   \n",
       "4    60% (4196)   [61% (1860) / 59% (2336)]       E -> D (522)   \n",
       "5     38% (2061)   [38% (817) / 38% (1244)]       E -> D (522)   \n",
       "6      4.8% (349)   [6% (159) / 4.1% (190)]             5' UTR   \n",
       "7    29% (2459)   [33% (1122) / 27% (1337)]             5' UTR   \n",
       "8    95% (9632)   [95% (4890) / 96% (4742)]  S -> [STOP] (338)   \n",
       "9    50% (5186)   [49% (2692) / 51% (2494)]  S -> [STOP] (338)   \n",
       "10   55% (3323)   [53% (1019) / 56% (2304)]       R -> Q (780)   \n",
       "11   60% (3707)   [59% (1126) / 60% (2581)]       R -> Q (780)   \n",
       "12   47% (3315)   [49% (1801) / 45% (1514)]                NaN   \n",
       "13   31% (2217)   [32% (1200) / 30% (1017)]                NaN   \n",
       "14   94% (9509)   [94% (5096) / 95% (4413)]       K -> E (132)   \n",
       "15   45% (5485)   [45% (3137) / 45% (2348)]       K -> E (132)   \n",
       "16                                      NaN                NaN   \n",
       "17                                      NaN                NaN   \n",
       "18                                      NaN                NaN   \n",
       "\n",
       "                                                        Annotation  \\\n",
       "0                                synonymous, coding sequence, exon   \n",
       "1                                synonymous, coding sequence, exon   \n",
       "2                missense, protein altering, coding sequence, exon   \n",
       "3                missense, protein altering, coding sequence, exon   \n",
       "4                missense, protein altering, coding sequence, exon   \n",
       "5                missense, protein altering, coding sequence, exon   \n",
       "6                                                     5' UTR, exon   \n",
       "7                                                     5' UTR, exon   \n",
       "8   stop gained, missense, protein altering, coding sequence, exon   \n",
       "9   stop gained, missense, protein altering, coding sequence, exon   \n",
       "10               missense, protein altering, coding sequence, exon   \n",
       "11               missense, protein altering, coding sequence, exon   \n",
       "12                                                          intron   \n",
       "13                                                          intron   \n",
       "14               missense, protein altering, coding sequence, exon   \n",
       "15               missense, protein altering, coding sequence, exon   \n",
       "16                                                             NaN   \n",
       "17                                                             NaN   \n",
       "18                                                             NaN   \n",
       "\n",
       "               Hint  \\\n",
       "0   primer position   \n",
       "1   primer position   \n",
       "2               NaN   \n",
       "3               NaN   \n",
       "4               NaN   \n",
       "5               NaN   \n",
       "6               NaN   \n",
       "7               NaN   \n",
       "8   primer position   \n",
       "9   primer position   \n",
       "10  primer position   \n",
       "11  primer position   \n",
       "12              NaN   \n",
       "13              NaN   \n",
       "14              NaN   \n",
       "15              NaN   \n",
       "16              NaN   \n",
       "17              NaN   \n",
       "18              NaN   \n",
       "\n",
       "                                                                      web Ref.  \\\n",
       "0                               rs151207425 (1000Genomes; ExAC; dbSNP; gnomAD)   \n",
       "1                               rs151207425 (1000Genomes; ExAC; dbSNP; gnomAD)   \n",
       "2   COSM1567768 (COSMIC), rs351855 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "3   COSM1567768 (COSMIC), rs351855 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "4                                                                          NaN   \n",
       "5                                                                          NaN   \n",
       "6                       rs41294870 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "7                       rs41294870 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "8                                                                          NaN   \n",
       "9                                                                          NaN   \n",
       "10                      rs35669708 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "11                      rs35669708 (1000Genomes; ClinVar; ExAC; dbSNP; gnomAD)   \n",
       "12                                                                         NaN   \n",
       "13                                                                         NaN   \n",
       "14                  rs747342068 (ClinVar; dbSNP; gnomAD), COSM1649370 (COSMIC)   \n",
       "15                  rs747342068 (ClinVar; dbSNP; gnomAD), COSM1649370 (COSMIC)   \n",
       "16                                                                         NaN   \n",
       "17                                                                         NaN   \n",
       "18                                                                         NaN   \n",
       "\n",
       "         c. HGVS      p. HGVS                              mut Entry  \\\n",
       "0      c.2118T>C     p.His706                            0/7 of 2382   \n",
       "1      c.2118T>C     p.His706                            0/7 of 2382   \n",
       "2      c.1162G>A  p.Gly388Arg                       225/1122 of 2861   \n",
       "3      c.1162G>A  p.Gly388Arg                       225/1122 of 2861   \n",
       "4      c.1566G>T  p.Glu522Asp                                    NaN   \n",
       "5      c.1566G>T  p.Glu522Asp                                    NaN   \n",
       "6       c.-10C>T          NaN  1/43 of 697 [c.-10C>T (NM_001130442)]   \n",
       "7       c.-10C>T          NaN  1/43 of 697 [c.-10C>T (NM_001130442)]   \n",
       "8      c.1013C>A  p.Ser338Ter                                    NaN   \n",
       "9      c.1013C>A  p.Ser338Ter                                    NaN   \n",
       "10     c.2339G>A  p.Arg780Gln                           0/15 of 1598   \n",
       "11     c.2339G>A  p.Arg780Gln                           0/15 of 1598   \n",
       "12  c.2285-38G>T          NaN                                    NaN   \n",
       "13  c.2285-38G>T          NaN                                    NaN   \n",
       "14      c.394A>G  p.Lys132Glu                            0/7 of 6835   \n",
       "15      c.394A>G  p.Lys132Glu                            0/7 of 6835   \n",
       "16           NaN          NaN                                    NaN   \n",
       "17           NaN          NaN                                    NaN   \n",
       "18           NaN          NaN                                    NaN   \n",
       "\n",
       "        mut Effect  TValidation  MValidation Weighting MutDB:Classification  \\\n",
       "0              NaN          NaN          NaN  distinct        likely benign   \n",
       "1              NaN          NaN          NaN  distinct        likely benign   \n",
       "2   Polymorphismus          NaN          NaN  distinct        likely benign   \n",
       "3   Polymorphismus          NaN          NaN  distinct        likely benign   \n",
       "4              NaN          NaN          NaN  distinct                  NaN   \n",
       "5              NaN          NaN          NaN  distinct                  NaN   \n",
       "6              NaN          NaN          NaN  distinct               benign   \n",
       "7              NaN          NaN          NaN  distinct               benign   \n",
       "8              NaN          NaN          NaN  distinct                  NaN   \n",
       "9              NaN          NaN          NaN  distinct                  NaN   \n",
       "10             NaN          NaN          NaN  distinct        likely benign   \n",
       "11             NaN          NaN          NaN  distinct        likely benign   \n",
       "12             NaN          NaN          NaN  distinct                  NaN   \n",
       "13             NaN          NaN          NaN  distinct                  NaN   \n",
       "14             NaN          NaN          NaN  distinct           pathogenic   \n",
       "15             NaN          NaN          NaN  distinct           pathogenic   \n",
       "16             NaN          NaN          NaN       NaN                  NaN   \n",
       "17             NaN          NaN          NaN       NaN                  NaN   \n",
       "18             NaN          NaN          NaN       NaN                  NaN   \n",
       "\n",
       "   MutDB:Has Weblinks  \\\n",
       "0                  no   \n",
       "1                  no   \n",
       "2                  no   \n",
       "3                  no   \n",
       "4                 NaN   \n",
       "5                 NaN   \n",
       "6                  no   \n",
       "7                  no   \n",
       "8                 NaN   \n",
       "9                 NaN   \n",
       "10                 no   \n",
       "11                 no   \n",
       "12                NaN   \n",
       "13                NaN   \n",
       "14                 no   \n",
       "15                 no   \n",
       "16                NaN   \n",
       "17                NaN   \n",
       "18                NaN   \n",
       "\n",
       "                                                                  ClinVitae:Classification  \\\n",
       "0                                                                                      NaN   \n",
       "1                                                                                      NaN   \n",
       "2                                                                               Pathogenic   \n",
       "3                                                                               Pathogenic   \n",
       "4                                                                                      NaN   \n",
       "5                                                                                      NaN   \n",
       "6                                                                      Benign,Not Provided   \n",
       "7                                                                      Benign,Not Provided   \n",
       "8                                                                                      NaN   \n",
       "9                                                                                      NaN   \n",
       "10                                                                                     NaN   \n",
       "11                                                                                     NaN   \n",
       "12                                                                                     NaN   \n",
       "13                                                                                     NaN   \n",
       "14  Variant of uncertain significance, likely pathogenic,Variant of uncertain significance   \n",
       "15  Variant of uncertain significance, likely pathogenic,Variant of uncertain significance   \n",
       "16                                                                                     NaN   \n",
       "17                                                                                     NaN   \n",
       "18                                                                                     NaN   \n",
       "\n",
       "           ClinVitae:Reported Classification  COSMIC:Pubmed_PMID  \\\n",
       "0                                        NaN                 NaN   \n",
       "1                                        NaN                 NaN   \n",
       "2                                 Pathogenic                 NaN   \n",
       "3                                 Pathogenic                 NaN   \n",
       "4                                        NaN                 NaN   \n",
       "5                                        NaN                 NaN   \n",
       "6                        Benign,not provided                 NaN   \n",
       "7                        Benign,not provided                 NaN   \n",
       "8                                        NaN                 NaN   \n",
       "9                                        NaN                 NaN   \n",
       "10                                       NaN                 NaN   \n",
       "11                                       NaN                 NaN   \n",
       "12                                       NaN                 NaN   \n",
       "13                                       NaN                 NaN   \n",
       "14  Likely pathogenic,Uncertain significance       1.041699e+119   \n",
       "15  Likely pathogenic,Uncertain significance       1.041699e+119   \n",
       "16                                       NaN                 NaN   \n",
       "17                                       NaN                 NaN   \n",
       "18                                       NaN                 NaN   \n",
       "\n",
       "                   ClinVar:Clinical Significance  ClinVar:ReviewStatus  \\\n",
       "0                                            NaN                   NaN   \n",
       "1                                            NaN                   NaN   \n",
       "2                                     Pathogenic                   NaN   \n",
       "3                                     Pathogenic                   NaN   \n",
       "4                                            NaN                   NaN   \n",
       "5                                            NaN                   NaN   \n",
       "6                                         Benign                   NaN   \n",
       "7                                         Benign                   NaN   \n",
       "8                                            NaN                   NaN   \n",
       "9                                            NaN                   NaN   \n",
       "10  Conflicting interpretations of pathogenicity                   NaN   \n",
       "11  Conflicting interpretations of pathogenicity                   NaN   \n",
       "12                                           NaN                   NaN   \n",
       "13                                           NaN                   NaN   \n",
       "14  Conflicting interpretations of pathogenicity                   NaN   \n",
       "15  Conflicting interpretations of pathogenicity                   NaN   \n",
       "16                                           NaN                   NaN   \n",
       "17                                           NaN                   NaN   \n",
       "18                                           NaN                   NaN   \n",
       "\n",
       "    1000Genomes:AF  1000Genomes:AVGPOST  1000Genomes:EUR_AF  dbSNP:ASS  \\\n",
       "0         0.000200                  NaN              0.0010        NaN   \n",
       "1         0.000200                  NaN              0.0010        NaN   \n",
       "2         0.299521                  NaN              0.2942        NaN   \n",
       "3         0.299521                  NaN              0.2942        NaN   \n",
       "4              NaN                  NaN                 NaN        NaN   \n",
       "5              NaN                  NaN                 NaN        NaN   \n",
       "6         0.035743                  NaN              0.0457        NaN   \n",
       "7         0.035743                  NaN              0.0457        NaN   \n",
       "8              NaN                  NaN                 NaN        NaN   \n",
       "9              NaN                  NaN                 NaN        NaN   \n",
       "10        0.003794                  NaN              0.0070        NaN   \n",
       "11        0.003794                  NaN              0.0070        NaN   \n",
       "12             NaN                  NaN                 NaN        NaN   \n",
       "13             NaN                  NaN                 NaN        NaN   \n",
       "14             NaN                  NaN                 NaN        NaN   \n",
       "15             NaN                  NaN                 NaN        NaN   \n",
       "16             NaN                  NaN                 NaN        NaN   \n",
       "17             NaN                  NaN                 NaN        NaN   \n",
       "18             NaN                  NaN                 NaN        NaN   \n",
       "\n",
       "   dbSNP:COMMON  dbSNP:DSS  \n",
       "0           NaN        NaN  \n",
       "1           NaN        NaN  \n",
       "2           yes        NaN  \n",
       "3           yes        NaN  \n",
       "4           NaN        NaN  \n",
       "5           NaN        NaN  \n",
       "6           yes        NaN  \n",
       "7           yes        NaN  \n",
       "8           NaN        NaN  \n",
       "9           NaN        NaN  \n",
       "10          NaN        NaN  \n",
       "11          NaN        NaN  \n",
       "12          NaN        NaN  \n",
       "13          NaN        NaN  \n",
       "14          NaN        NaN  \n",
       "15          NaN        NaN  \n",
       "16          NaN        NaN  \n",
       "17          NaN        NaN  \n",
       "18          NaN        NaN  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(os.path.join(workdir, \"testdata/test_mutations_hg19.xlsx\"))\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62f756d7",
   "metadata": {},
   "source": [
    "There are a lot of columns and information right now. The last rows are leftovers from the excel transformation so we get rid of them with iloc. To make things easy, we also remove every additional column right now and keep only the columns needed for annotation.\n",
    "The coords are somehow stored in the `Pos.` column and the mutation in the `Nuc Change` column. So do all this with `iloc`.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "2a0e65e7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:42:59.300843Z",
     "start_time": "2022-02-06T11:42:59.284699Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pos.</th>\n",
       "      <th>Nuc Change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>C -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>C -&gt; A (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>A -&gt; G (het)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Pos.     Nuc Change\n",
       "0   103 (2118)   [chr5:g.176523707 (hg19)]   T -> C (het)\n",
       "1   103 (2118)   [chr5:g.176523707 (hg19)]   T -> C (het)\n",
       "2   105 (1162)   [chr5:g.176520243 (hg19)]  G -> A (homo)\n",
       "3   105 (1162)   [chr5:g.176520243 (hg19)]  G -> A (homo)\n",
       "4    47 (1566)   [chr5:g.176522377 (hg19)]   G -> T (het)\n",
       "5    47 (1566)   [chr5:g.176522377 (hg19)]   G -> T (het)\n",
       "6             44   [chr11:g.534332 (hg19)]   C -> T (het)\n",
       "7             44   [chr11:g.534332 (hg19)]   G -> A (het)\n",
       "8   374 (1013)   [chr19:g.10602565 (hg19)]   C -> A (het)\n",
       "9   374 (1013)   [chr19:g.10602565 (hg19)]   G -> T (het)\n",
       "10  134 (2339)   [chr1:g.156851382 (hg19)]   G -> A (het)\n",
       "11  134 (2339)   [chr1:g.156851382 (hg19)]   G -> A (het)\n",
       "12         -38   [chr10:g.43613783 (hg19)]   G -> T (het)\n",
       "13         -38   [chr10:g.43613783 (hg19)]   G -> T (het)\n",
       "14     19 (394)   [chr17:g.7578536 (hg19)]   A -> G (het)\n",
       "15     19 (394)   [chr17:g.7578536 (hg19)]   T -> C (het)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.iloc[:16, [4,6]]\n",
    "# df = df.iloc[:15, :]\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a40ad04",
   "metadata": {},
   "source": [
    "### wrangle\n",
    "Now, we extract the relevant information with str.extract and some smart regex. This will be different everytime and if things get too complicated consider chaining several extracts and even to modify the source file itself. Try to be efficient here!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "ec9fcf0e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:43:00.381268Z",
     "start_time": "2022-02-06T11:43:00.356575Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pos.</th>\n",
       "      <th>Nuc Change</th>\n",
       "      <th>Start</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>End</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>176523707</td>\n",
       "      <td>chr5</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>176523707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>103 (2118)   [chr5:g.176523707 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>176523707</td>\n",
       "      <td>chr5</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>176523707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "      <td>176520243</td>\n",
       "      <td>chr5</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>176520243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>105 (1162)   [chr5:g.176520243 (hg19)]</td>\n",
       "      <td>G -&gt; A (homo)</td>\n",
       "      <td>176520243</td>\n",
       "      <td>chr5</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>176520243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>176522377</td>\n",
       "      <td>chr5</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>176522377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>47 (1566)   [chr5:g.176522377 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>176522377</td>\n",
       "      <td>chr5</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>176522377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>C -&gt; T (het)</td>\n",
       "      <td>534332</td>\n",
       "      <td>chr11</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>534332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>44   [chr11:g.534332 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>534332</td>\n",
       "      <td>chr11</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>534332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>C -&gt; A (het)</td>\n",
       "      <td>10602565</td>\n",
       "      <td>chr19</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>10602565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>374 (1013)   [chr19:g.10602565 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>10602565</td>\n",
       "      <td>chr19</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>10602565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>156851382</td>\n",
       "      <td>chr1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>156851382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>134 (2339)   [chr1:g.156851382 (hg19)]</td>\n",
       "      <td>G -&gt; A (het)</td>\n",
       "      <td>156851382</td>\n",
       "      <td>chr1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>156851382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>43613783</td>\n",
       "      <td>chr10</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>43613783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>-38   [chr10:g.43613783 (hg19)]</td>\n",
       "      <td>G -&gt; T (het)</td>\n",
       "      <td>43613783</td>\n",
       "      <td>chr10</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>43613783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>A -&gt; G (het)</td>\n",
       "      <td>7578536</td>\n",
       "      <td>chr17</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>7578536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>19 (394)   [chr17:g.7578536 (hg19)]</td>\n",
       "      <td>T -&gt; C (het)</td>\n",
       "      <td>7578536</td>\n",
       "      <td>chr17</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7578536</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Pos.     Nuc Change      Start    Chr  \\\n",
       "0   103 (2118)   [chr5:g.176523707 (hg19)]   T -> C (het)  176523707   chr5   \n",
       "1   103 (2118)   [chr5:g.176523707 (hg19)]   T -> C (het)  176523707   chr5   \n",
       "2   105 (1162)   [chr5:g.176520243 (hg19)]  G -> A (homo)  176520243   chr5   \n",
       "3   105 (1162)   [chr5:g.176520243 (hg19)]  G -> A (homo)  176520243   chr5   \n",
       "4    47 (1566)   [chr5:g.176522377 (hg19)]   G -> T (het)  176522377   chr5   \n",
       "5    47 (1566)   [chr5:g.176522377 (hg19)]   G -> T (het)  176522377   chr5   \n",
       "6             44   [chr11:g.534332 (hg19)]   C -> T (het)     534332  chr11   \n",
       "7             44   [chr11:g.534332 (hg19)]   G -> A (het)     534332  chr11   \n",
       "8   374 (1013)   [chr19:g.10602565 (hg19)]   C -> A (het)   10602565  chr19   \n",
       "9   374 (1013)   [chr19:g.10602565 (hg19)]   G -> T (het)   10602565  chr19   \n",
       "10  134 (2339)   [chr1:g.156851382 (hg19)]   G -> A (het)  156851382   chr1   \n",
       "11  134 (2339)   [chr1:g.156851382 (hg19)]   G -> A (het)  156851382   chr1   \n",
       "12         -38   [chr10:g.43613783 (hg19)]   G -> T (het)   43613783  chr10   \n",
       "13         -38   [chr10:g.43613783 (hg19)]   G -> T (het)   43613783  chr10   \n",
       "14     19 (394)   [chr17:g.7578536 (hg19)]   A -> G (het)    7578536  chr17   \n",
       "15     19 (394)   [chr17:g.7578536 (hg19)]   T -> C (het)    7578536  chr17   \n",
       "\n",
       "   Ref Alt        End  \n",
       "0    T   C  176523707  \n",
       "1    T   C  176523707  \n",
       "2    G   A  176520243  \n",
       "3    G   A  176520243  \n",
       "4    G   T  176522377  \n",
       "5    G   T  176522377  \n",
       "6    C   T     534332  \n",
       "7    G   A     534332  \n",
       "8    C   A   10602565  \n",
       "9    G   T   10602565  \n",
       "10   G   A  156851382  \n",
       "11   G   A  156851382  \n",
       "12   G   T   43613783  \n",
       "13   G   T   43613783  \n",
       "14   A   G    7578536  \n",
       "15   T   C    7578536  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[:,[\"Start\", \"Chr\"]] = df['Pos.'].str.extract(r\"(?P<Chr>chr[0-9]+):g\\.(?P<Start>[0-9]+)\")\n",
    "df.loc[:,[\"Ref\", \"Alt\"]] = df['Nuc Change'].str.extract(r\"(?P<Ref>[ACTG]) -> (?P<Alt>[ACTG])\")\n",
    "df.loc[:, \"End\"] = df['Start']\n",
    "# df = df.loc[:,[\"Chr\", \"Start\", \"End\", \"Ref\", \"Alt\"]]\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f21f0e09",
   "metadata": {},
   "source": [
    "### convert to hg38\n",
    "Although there are some tools that do that automatically, I still use the USCS website for conversion. For that, the coords have to be converted to the bed-format `Chr:Start-End`. This is always tedious and I use the helpers provided by the repo.\n",
    "Here are the steps:\n",
    "+ create the coords using `pos2bed` helper. Set option `as_string=True` and print the output for direct copying to the browser as it removes the index\n",
    "+ copy to the website and create the conversion\n",
    "+ reintroduce the hg38 coords into the file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "97020e21",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:43:03.586673Z",
     "start_time": "2022-02-06T11:43:03.574667Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5:176523707-176523707\n",
      "chr5:176523707-176523707\n",
      "chr5:176520243-176520243\n",
      "chr5:176520243-176520243\n",
      "chr5:176522377-176522377\n",
      "chr5:176522377-176522377\n",
      "     chr11:534332-534332\n",
      "     chr11:534332-534332\n",
      " chr19:10602565-10602565\n",
      " chr19:10602565-10602565\n",
      "chr1:156851382-156851382\n",
      "chr1:156851382-156851382\n",
      " chr10:43613783-43613783\n",
      " chr10:43613783-43613783\n",
      "   chr17:7578536-7578536\n",
      "   chr17:7578536-7578536\n"
     ]
    }
   ],
   "source": [
    "from pyseq_utils import pos2bed\n",
    "print(pos2bed(df, as_string=True))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "222cda65",
   "metadata": {},
   "source": [
    "+ Now copy the coords to the clipboard and paste into the [uscs liftover website](https://genome.ucsc.edu/cgi-bin/hgLiftOver)\n",
    "+ click on `View conversions` and retrieve the file from your download folder. Should be something like `hglft_genome....be`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "a0cc08ad",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:43:04.557599Z",
     "start_time": "2022-02-06T11:43:04.541988Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hg38</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5:177096706-177096706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5:177096706-177096706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5:177093242-177093242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5:177093242-177093242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5:177095376-177095376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5:177095376-177095376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr11:534332-534332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr11:534332-534332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr19:10491889-10491889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr19:10491889-10491889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1:156881590-156881590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1:156881590-156881590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10:43118335-43118335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10:43118335-43118335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr17:7675218-7675218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr17:7675218-7675218</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        hg38\n",
       "0   chr5:177096706-177096706\n",
       "1   chr5:177096706-177096706\n",
       "2   chr5:177093242-177093242\n",
       "3   chr5:177093242-177093242\n",
       "4   chr5:177095376-177095376\n",
       "5   chr5:177095376-177095376\n",
       "6        chr11:534332-534332\n",
       "7        chr11:534332-534332\n",
       "8    chr19:10491889-10491889\n",
       "9    chr19:10491889-10491889\n",
       "10  chr1:156881590-156881590\n",
       "11  chr1:156881590-156881590\n",
       "12   chr10:43118335-43118335\n",
       "13   chr10:43118335-43118335\n",
       "14     chr17:7675218-7675218\n",
       "15     chr17:7675218-7675218"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38 = pd.read_csv(os.path.join(home, \"Downloads/hglft_genome_1181f_fb3d20.bed\"), sep=\"\\t\", names=['hg38'])\n",
    "hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e23c2896",
   "metadata": {},
   "source": [
    "`bed2pos`reconverts the column into Chr Start End that can be reinserted into the dataframe. This only works by index, so make sure that your original df indices have not changed and are in a reset state!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "bbbd8189",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:43:05.780515Z",
     "start_time": "2022-02-06T11:43:05.758882Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177093242</td>\n",
       "      <td>177093242</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177093242</td>\n",
       "      <td>177093242</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177095376</td>\n",
       "      <td>177095376</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177095376</td>\n",
       "      <td>177095376</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr11</td>\n",
       "      <td>534332</td>\n",
       "      <td>534332</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr11</td>\n",
       "      <td>534332</td>\n",
       "      <td>534332</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr19</td>\n",
       "      <td>10491889</td>\n",
       "      <td>10491889</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr19</td>\n",
       "      <td>10491889</td>\n",
       "      <td>10491889</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156881590</td>\n",
       "      <td>156881590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156881590</td>\n",
       "      <td>156881590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43118335</td>\n",
       "      <td>43118335</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43118335</td>\n",
       "      <td>43118335</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675218</td>\n",
       "      <td>7675218</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675218</td>\n",
       "      <td>7675218</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt\n",
       "0    chr5  177096706  177096706   T   C\n",
       "1    chr5  177096706  177096706   T   C\n",
       "2    chr5  177093242  177093242   G   A\n",
       "3    chr5  177093242  177093242   G   A\n",
       "4    chr5  177095376  177095376   G   T\n",
       "5    chr5  177095376  177095376   G   T\n",
       "6   chr11     534332     534332   C   T\n",
       "7   chr11     534332     534332   G   A\n",
       "8   chr19   10491889   10491889   C   A\n",
       "9   chr19   10491889   10491889   G   T\n",
       "10   chr1  156881590  156881590   G   A\n",
       "11   chr1  156881590  156881590   G   A\n",
       "12  chr10   43118335   43118335   G   T\n",
       "13  chr10   43118335   43118335   G   T\n",
       "14  chr17    7675218    7675218   A   G\n",
       "15  chr17    7675218    7675218   T   C"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyseq_utils import bed2pos\n",
    "df.loc[:, [\"Chr\", \"Start\", \"End\"]] = bed2pos(hg38['hg38'])\n",
    "df = df.loc[:, ['Chr', 'Start', 'End', 'Ref', 'Alt']]\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2cfc618a",
   "metadata": {},
   "source": [
    "### annotate data with annovar\n",
    "Annovar is a very comprehensive tool for annotation of mutations using various databases. See [here](https://annovar.openbioinformatics.org/en/latest/) for all you need to know. I provide a static file for genomic annotation that contains (among others) updated databases for use with annovar.\n",
    "For downloading the static files, do this from the root folder:\n",
    "+ `$ . setup/download_static.sh <path-to-static-folder>  # provide a folder for downloading and expanding ~40GB of data`\n",
    "+ for annovar to work, the path to the static folder has to be set in an environment variable STATIC: \n",
    "    * `$ export STATIC=<path-to-static-folder>`. \n",
    "    * For making this permanent, copy this line to your .bash_profile file in your HOME folder:\n",
    "    * (`$ echo \"export STATIC=<path-to-static-folder>\" >> \"${HOME}/.bash_profile`)\n",
    "\n",
    "This will take some time as it is downloading and unpacking a 40GB file!!\n",
    "\n",
    "Next, you have to configure the annovar_config.yaml file, giving it\n",
    "   + the absolute path to the annovar folder sitting in this repo at ./code/anno2019. This depends on where your repo is residing.\n",
    "   + the path in the STATIC folder to the annovar database (only change this if you moved the humandb folder somewhere else relative to the static folder)\n",
    "   + a list of databases from the annovar database to use for populating the mutations\n",
    "        * see the list of currently stored databases like so:`$ ls ${STATIC}/hg38/annotation/annovar/humandb/*.txt`\n",
    "        * if you want the database hg38_icgc29.txt to be used, just list \"icgc29\" in the yaml file\n",
    "\n",
    "Now, can run the annovar tool (run_annovar is a convenience wrapper around the command line tool written in perl), so you only have to worry once about providing the absolute path to the annovar code folder. You can see the large \n",
    "This will take some time depending on the list and size of both the mutations and the number and size of the databases used. You can see the long command line call in the cell output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "8e8e90e4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:43:18.198080Z",
     "start_time": "2022-02-06T11:43:07.166241Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m$ perl /Users/martinszyska/Sites/Bio/pyseq/code/anno2019/table_annovar.pl -buildver hg38 --maxgenethread 10 --thread 10 --protocol ensGene34,cytoBand,gnomad30,dbSNP154,cosmic95,icgc29,clinvar2021 --operation g,r,f,f,f,f,f -nastring \".\" --remove --outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb\u001b[0m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: the --polish argument is set ON automatically (use --nopolish to change this behavior)\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=ensGene34\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype ensGene34 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34 -exonsort -nofirstcodondel /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -maxgenethread 10>\n",
      "NOTICE: Output files are written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34.variant_function, /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34.exonic_variant_function\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: threading is disabled for gene-based annotation on file with less than 1000000 input lines\n",
      "NOTICE: Reading gene annotation from /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_ensGene34.txt ... Done with 100502 transcripts (including 35147 without coding sequence annotation) for 46639 unique genes\n",
      "NOTICE: Processing next batch with 16 unique variants in 16 input lines\n",
      "NOTICE: Reading FASTA sequences from /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_ensGene34Mrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 385 sequences will be ignored due to lack of correct ORF annotation\n",
      "\n",
      "---------------------------------------------------------------------------------------\n",
      "WARNING: 2 exonic SNPs have WRONG reference alleles specified in your input file!\n",
      "WARNING: An example input line is <chr17\t7675218\t7675218\tA\tG>\n",
      "WARNING: ANNOVAR can still annotate exonic_variant_function for the mutation correctly!\n",
      "WARNING: you may have used wrong -buildver, or specified incorrect reference allele, or used outdated mRNA FASTA file!\n",
      "---------------------------------------------------------------------------------------\n",
      "\n",
      "\n",
      "NOTICE: Running with system command <coding_change.pl  /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34.exonic_variant_function.orig /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_ensGene34.txt /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_ensGene34Mrna.fa -alltranscript -out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34.fa -newevf /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.ensGene34.exonic_variant_function>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=r protocol=cytoBand\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -regionanno -dbtype cytoBand -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10>\n",
      "NOTICE: Output file is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_cytoBand\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 4 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 4 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 4 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 4 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "NOTICE: Reading annotation database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cytoBand.txt ... Done with 1433 regions\n",
      "NOTICE: Finished region-based annotation on 0 genetic variants\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=gnomad30\n",
      "NOTICE: Finished reading 1 column headers for '-dbtype gnomad30'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype gnomad30 -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -otherinfo>\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_gnomad30_dropped, and output file with other variants is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_gnomad30_filtered\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314716 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_gnomad30.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314716 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_gnomad30.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314716 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_gnomad30.txt...Done\n",
      "NOTICE: Processing next batch with 2 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314716 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_gnomad30.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=dbSNP154\n",
      "NOTICE: Finished reading 3 column headers for '-dbtype dbSNP154'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype dbSNP154 -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -otherinfo>\n",
      "NOTICE: the --dbtype dbSNP154 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_dbSNP154_dropped, and output file with other variants is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_dbSNP154_filtered\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Processing next batch with 2 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314742 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_dbSNP154.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314742 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_dbSNP154.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314742 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_dbSNP154.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 314742 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_dbSNP154.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=cosmic95\n",
      "NOTICE: Finished reading 5 column headers for '-dbtype cosmic95'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype cosmic95 -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -otherinfo>\n",
      "NOTICE: the --dbtype cosmic95 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_cosmic95_dropped, and output file with other variants is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_cosmic95_filtered\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 1144806 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cosmic95.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 1144806 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cosmic95.txt...Done\n",
      "NOTICE: Processing next batch with 2 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 1144806 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cosmic95.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 1144806 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_cosmic95.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=icgc29\n",
      "NOTICE: Finished reading 2 column headers for '-dbtype icgc29'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype icgc29 -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -otherinfo>\n",
      "NOTICE: the --dbtype icgc29 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_icgc29_dropped, and output file with other variants is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_icgc29_filtered\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n",
      "NOTICE: Processing next batch with 2 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 2836799 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_icgc29.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 2836799 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_icgc29.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 2836799 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_icgc29.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 2836799 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_icgc29.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar2021\n",
      "NOTICE: Finished reading 3 column headers for '-dbtype clinvar2021'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar2021 -buildver hg38 -outfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb -thread 10 -otherinfo>\n",
      "NOTICE: the --dbtype clinvar2021 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_clinvar2021_dropped, and output file with other variants is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_clinvar2021_filtered\n",
      "NOTICE: the queryfile /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/anno_file.csv contains 32 lines\n",
      "NOTICE: Creating new threads for query line 1 to 4\n",
      "NOTICE: Creating new threads for query line 5 to 8\n",
      "NOTICE: Creating new threads for query line 9 to 12\n",
      "NOTICE: Creating new threads for query line 13 to 16\n",
      "NOTICE: Creating new threads for query line 17 to 20\n",
      "NOTICE: Creating new threads for query line 21 to 24\n",
      "NOTICE: Creating new threads for query line 25 to 28\n",
      "NOTICE: Creating new threads for query line 29 to 32\n",
      "NOTICE: Creating new threads for query line 33 to 32\n",
      "NOTICE: Creating new threads for query line 37 to 32\n",
      "NOTICE: Processing next batch with 2 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 76449 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_clinvar2021.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 76449 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_clinvar2021.txt...Done\n",
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 76449 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_clinvar2021.txt...Done\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr      object\n",
      "Start     int64\n",
      "End       int64\n",
      "Ref      object\n",
      "Alt      object\n",
      "dtype: object Chr                     object\n",
      "Start                    int64\n",
      "End                      int64\n",
      "Ref                     object\n",
      "Alt                     object\n",
      "Func.ensGene34          object\n",
      "Gene.ensGene34          object\n",
      "GeneDetail.ensGene34    object\n",
      "ExonicFunc.ensGene34    object\n",
      "AAChange.ensGene34      object\n",
      "cytoBand                object\n",
      "gnomAD_exome_ALL        object\n",
      "dbSNP154_ID             object\n",
      "dbSNP154_AltFreq        object\n",
      "dbSNP154_DB             object\n",
      "Mut_ID                  object\n",
      "Mut_ID_old              object\n",
      "count                   object\n",
      "type                    object\n",
      "MutStatus               object\n",
      "icgc29_ID               object\n",
      "icgc29_Affected         object\n",
      "CLNALLELEID             object\n",
      "CLNDN                   object\n",
      "CLNSIG                  object\n",
      "dtype: object\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Processing next batch with 3 unique variants in 4 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 76449 and the number of bins to be scanned is 2\n",
      "NOTICE: Scanning filter database /Users/martinszyska/Dropbox/Icke/Work/static/hg38/annotation/annovar/humandb/hg38_clinvar2021.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /Users/martinszyska/Sites/Bio/pyseq/nb/anno_temp/annovar_out.hg38_multianno.txt\n"
     ]
    }
   ],
   "source": [
    "from anno import run_annovar\n",
    "config_file = \"../configs/annovar_config.yaml\"\n",
    "                         \n",
    "df_anno = run_annovar(df, config_file, threads=10, cleanup=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e6eb7fa",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-05T23:35:13.655571Z",
     "start_time": "2022-02-05T23:35:13.636232Z"
    }
   },
   "source": [
    "### The annovar result\n",
    "You can inspect the output and see that, depending on the used databases you have several new columns in your output dataframe\n",
    "But first, it would be wise to remove some of the columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "031924b7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:45:22.039439Z",
     "start_time": "2022-02-06T11:45:22.016727Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.ensGene34</th>\n",
       "      <th>Gene.ensGene34</th>\n",
       "      <th>GeneDetail.ensGene34</th>\n",
       "      <th>ExonicFunc.ensGene34</th>\n",
       "      <th>AAChange.ensGene34</th>\n",
       "      <th>cytoBand</th>\n",
       "      <th>gnomAD_exome_ALL</th>\n",
       "      <th>dbSNP154_ID</th>\n",
       "      <th>dbSNP154_AltFreq</th>\n",
       "      <th>dbSNP154_DB</th>\n",
       "      <th>Mut_ID</th>\n",
       "      <th>Mut_ID_old</th>\n",
       "      <th>count</th>\n",
       "      <th>type</th>\n",
       "      <th>MutStatus</th>\n",
       "      <th>icgc29_ID</th>\n",
       "      <th>icgc29_Affected</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>FGFR4:ENST00000393637.5:exon14:c.T1998C:p.H666H,FGFR4:ENST00000292408.9:exon16:c.T2118C:p.H706H,FGFR4:ENST00000393648.6:exon16:c.T1914C:p.H638H,FGFR4:ENST00000502906.5:exon16:c.T2118C:p.H706H</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>rs151207425</td>\n",
       "      <td>0.0001997</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>FGFR4:ENST00000393637.5:exon14:c.T1998C:p.H666H,FGFR4:ENST00000292408.9:exon16:c.T2118C:p.H706H,FGFR4:ENST00000393648.6:exon16:c.T1914C:p.H638H,FGFR4:ENST00000502906.5:exon16:c.T2118C:p.H706H</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>rs151207425</td>\n",
       "      <td>0.0001997</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177093242</td>\n",
       "      <td>177093242</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>FGFR4:ENST00000292408.9:exon9:c.G1162A:p.G388R,FGFR4:ENST00000502906.5:exon9:c.G1162A:p.G388R</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.2632</td>\n",
       "      <td>rs351855</td>\n",
       "      <td>0.2995</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>COSV52800825</td>\n",
       "      <td>COSM1567768</td>\n",
       "      <td>128</td>\n",
       "      <td>4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...</td>\n",
       "      <td>ReportedElsewhere</td>\n",
       "      <td>MU153762</td>\n",
       "      <td>2/170</td>\n",
       "      <td>31365</td>\n",
       "      <td>See_cases|Cancer_progression_and_tumor_cell_motility</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chr      Start        End Ref Alt Func.ensGene34 Gene.ensGene34  \\\n",
       "0  chr5  177096706  177096706   T   C         exonic          FGFR4   \n",
       "1  chr5  177096706  177096706   T   C         exonic          FGFR4   \n",
       "2  chr5  177093242  177093242   G   A         exonic          FGFR4   \n",
       "\n",
       "  GeneDetail.ensGene34 ExonicFunc.ensGene34  \\\n",
       "0                    .       synonymous SNV   \n",
       "1                    .       synonymous SNV   \n",
       "2                    .    nonsynonymous SNV   \n",
       "\n",
       "                                                                                                                                                                                AAChange.ensGene34  \\\n",
       "0  FGFR4:ENST00000393637.5:exon14:c.T1998C:p.H666H,FGFR4:ENST00000292408.9:exon16:c.T2118C:p.H706H,FGFR4:ENST00000393648.6:exon16:c.T1914C:p.H638H,FGFR4:ENST00000502906.5:exon16:c.T2118C:p.H706H   \n",
       "1  FGFR4:ENST00000393637.5:exon14:c.T1998C:p.H666H,FGFR4:ENST00000292408.9:exon16:c.T2118C:p.H706H,FGFR4:ENST00000393648.6:exon16:c.T1914C:p.H638H,FGFR4:ENST00000502906.5:exon16:c.T2118C:p.H706H   \n",
       "2                                                                                                    FGFR4:ENST00000292408.9:exon9:c.G1162A:p.G388R,FGFR4:ENST00000502906.5:exon9:c.G1162A:p.G388R   \n",
       "\n",
       "  cytoBand gnomAD_exome_ALL  dbSNP154_ID dbSNP154_AltFreq  dbSNP154_DB  \\\n",
       "0   5q35.2           0.0003  rs151207425        0.0001997  1000Genomes   \n",
       "1   5q35.2           0.0003  rs151207425        0.0001997  1000Genomes   \n",
       "2   5q35.2           0.2632     rs351855           0.2995  1000Genomes   \n",
       "\n",
       "         Mut_ID   Mut_ID_old count  \\\n",
       "0             .            .     .   \n",
       "1             .            .     .   \n",
       "2  COSV52800825  COSM1567768   128   \n",
       "\n",
       "                                                                                                                                                                                                      type  \\\n",
       "0                                                                                                                                                                                                        .   \n",
       "1                                                                                                                                                                                                        .   \n",
       "2  4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...   \n",
       "\n",
       "           MutStatus icgc29_ID icgc29_Affected CLNALLELEID  \\\n",
       "0                  .         .               .           .   \n",
       "1                  .         .               .           .   \n",
       "2  ReportedElsewhere  MU153762           2/170       31365   \n",
       "\n",
       "                                                  CLNDN  \\\n",
       "0                                                     .   \n",
       "1                                                     .   \n",
       "2  See_cases|Cancer_progression_and_tumor_cell_motility   \n",
       "\n",
       "                   CLNSIG  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2  Uncertain_significance  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_anno[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "85780f0d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-02-06T11:47:22.193558Z",
     "start_time": "2022-02-06T11:47:22.168840Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Chr', 'Start', 'End', 'Ref', 'Alt', 'Func.ensGene34', 'Gene.ensGene34',\n",
       "       'ExonicFunc.ensGene34', 'cytoBand', 'gnomAD_exome_ALL',\n",
       "       'dbSNP154_AltFreq', 'Mut_ID', 'count', 'type', 'icgc29_ID',\n",
       "       'icgc29_Affected', 'CLNSIG'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.ensGene34</th>\n",
       "      <th>Gene.ensGene34</th>\n",
       "      <th>ExonicFunc.ensGene34</th>\n",
       "      <th>cytoBand</th>\n",
       "      <th>gnomAD_exome_ALL</th>\n",
       "      <th>dbSNP154_AltFreq</th>\n",
       "      <th>Mut_ID</th>\n",
       "      <th>count</th>\n",
       "      <th>type</th>\n",
       "      <th>icgc29_ID</th>\n",
       "      <th>icgc29_Affected</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0001997</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177096706</td>\n",
       "      <td>177096706</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.0001997</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177093242</td>\n",
       "      <td>177093242</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.2632</td>\n",
       "      <td>0.2995</td>\n",
       "      <td>COSV52800825</td>\n",
       "      <td>128</td>\n",
       "      <td>4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...</td>\n",
       "      <td>MU153762</td>\n",
       "      <td>2/170</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177093242</td>\n",
       "      <td>177093242</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>0.2632</td>\n",
       "      <td>0.2995</td>\n",
       "      <td>COSV52800825</td>\n",
       "      <td>128</td>\n",
       "      <td>4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...</td>\n",
       "      <td>MU153762</td>\n",
       "      <td>2/170</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177095376</td>\n",
       "      <td>177095376</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>COSV99450816</td>\n",
       "      <td>4</td>\n",
       "      <td>4x(adenocarcinoma@lung)</td>\n",
       "      <td>MU131464768</td>\n",
       "      <td>1/516</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr5</td>\n",
       "      <td>177095376</td>\n",
       "      <td>177095376</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>FGFR4</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>5q35.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>COSV99450816</td>\n",
       "      <td>4</td>\n",
       "      <td>4x(adenocarcinoma@lung)</td>\n",
       "      <td>MU131464768</td>\n",
       "      <td>1/516</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr11</td>\n",
       "      <td>534332</td>\n",
       "      <td>534332</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>.</td>\n",
       "      <td>11p15.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr11</td>\n",
       "      <td>534332</td>\n",
       "      <td>534332</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>HRAS</td>\n",
       "      <td>.</td>\n",
       "      <td>11p15.5</td>\n",
       "      <td>0.0365</td>\n",
       "      <td>0.03574</td>\n",
       "      <td>COSV54240784</td>\n",
       "      <td>10</td>\n",
       "      <td>10x(haemangioblastoma@blood_vessel)</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr19</td>\n",
       "      <td>10491889</td>\n",
       "      <td>10491889</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>19p13.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>COSV50276845</td>\n",
       "      <td>6</td>\n",
       "      <td>2x(ER-PR-positive_carcinoma@breast)+2x(adenocarcinoma@gallbladder)+2x(squamous_cell_carcinoma@lung)</td>\n",
       "      <td>MU129512036</td>\n",
       "      <td>1/531</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr19</td>\n",
       "      <td>10491889</td>\n",
       "      <td>10491889</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>19p13.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156881590</td>\n",
       "      <td>156881590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>1q23.1</td>\n",
       "      <td>0.0043</td>\n",
       "      <td>0.003794</td>\n",
       "      <td>COSV62324745</td>\n",
       "      <td>12</td>\n",
       "      <td>4x(atypical@meninges)+4x(ductal_carcinoma@breast)+4x(plasma_cell_myeloma@haematopoietic_and_lymphoid_tissue)</td>\n",
       "      <td>MU66190450</td>\n",
       "      <td>1/569</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>156881590</td>\n",
       "      <td>156881590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>1q23.1</td>\n",
       "      <td>0.0043</td>\n",
       "      <td>0.003794</td>\n",
       "      <td>COSV62324745</td>\n",
       "      <td>12</td>\n",
       "      <td>4x(atypical@meninges)+4x(ductal_carcinoma@breast)+4x(plasma_cell_myeloma@haematopoietic_and_lymphoid_tissue)</td>\n",
       "      <td>MU66190450</td>\n",
       "      <td>1/569</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43118335</td>\n",
       "      <td>43118335</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>RET</td>\n",
       "      <td>.</td>\n",
       "      <td>10q11.21</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43118335</td>\n",
       "      <td>43118335</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>RET</td>\n",
       "      <td>.</td>\n",
       "      <td>10q11.21</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675218</td>\n",
       "      <td>7675218</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TP53</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>17p13.1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>COSV52708494</td>\n",
       "      <td>437</td>\n",
       "      <td>38x(B_cell_prolymphocytic_leukaemia@haematopoietic_and_lymphoid_tissue)+38x(acute_myeloid_leukaemia@haematopoietic_and_lymphoid_tissue)+19x(acute_myeloid_leukaemia_associated_with_MDS@haematopoiet...</td>\n",
       "      <td>MU1681389</td>\n",
       "      <td>1/16</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675218</td>\n",
       "      <td>7675218</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TP53</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>17p13.1</td>\n",
       "      <td>.</td>\n",
       "      <td>8.327e-06</td>\n",
       "      <td>COSV52689323</td>\n",
       "      <td>855</td>\n",
       "      <td>19x(ER-positive_carcinoma@breast)+19x(adenocarcinoma@bile_duct)+19x(adenocarcinoma@caecum)+19x(adenocarcinoma@colon)+19x(adenocarcinoma@gallbladder)+19x(adenocarcinoma@large_intestine)+57x(adenoca...</td>\n",
       "      <td>MU36255</td>\n",
       "      <td>1/241</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt Func.ensGene34 Gene.ensGene34  \\\n",
       "0    chr5  177096706  177096706   T   C         exonic          FGFR4   \n",
       "1    chr5  177096706  177096706   T   C         exonic          FGFR4   \n",
       "2    chr5  177093242  177093242   G   A         exonic          FGFR4   \n",
       "3    chr5  177093242  177093242   G   A         exonic          FGFR4   \n",
       "4    chr5  177095376  177095376   G   T         exonic          FGFR4   \n",
       "5    chr5  177095376  177095376   G   T         exonic          FGFR4   \n",
       "6   chr11     534332     534332   C   T           UTR5           HRAS   \n",
       "7   chr11     534332     534332   G   A           UTR5           HRAS   \n",
       "8   chr19   10491889   10491889   C   A         exonic          KEAP1   \n",
       "9   chr19   10491889   10491889   G   T         exonic          KEAP1   \n",
       "10   chr1  156881590  156881590   G   A         exonic          NTRK1   \n",
       "11   chr1  156881590  156881590   G   A         exonic          NTRK1   \n",
       "12  chr10   43118335   43118335   G   T       intronic            RET   \n",
       "13  chr10   43118335   43118335   G   T       intronic            RET   \n",
       "14  chr17    7675218    7675218   A   G         exonic           TP53   \n",
       "15  chr17    7675218    7675218   T   C         exonic           TP53   \n",
       "\n",
       "   ExonicFunc.ensGene34  cytoBand gnomAD_exome_ALL dbSNP154_AltFreq  \\\n",
       "0        synonymous SNV    5q35.2           0.0003        0.0001997   \n",
       "1        synonymous SNV    5q35.2           0.0003        0.0001997   \n",
       "2     nonsynonymous SNV    5q35.2           0.2632           0.2995   \n",
       "3     nonsynonymous SNV    5q35.2           0.2632           0.2995   \n",
       "4     nonsynonymous SNV    5q35.2                .                .   \n",
       "5     nonsynonymous SNV    5q35.2                .                .   \n",
       "6                     .   11p15.5                .                .   \n",
       "7                     .   11p15.5           0.0365          0.03574   \n",
       "8     nonsynonymous SNV   19p13.2                .                .   \n",
       "9              stopgain   19p13.2                .                .   \n",
       "10    nonsynonymous SNV    1q23.1           0.0043         0.003794   \n",
       "11    nonsynonymous SNV    1q23.1           0.0043         0.003794   \n",
       "12                    .  10q11.21                .                .   \n",
       "13                    .  10q11.21                .                .   \n",
       "14    nonsynonymous SNV   17p13.1                .                .   \n",
       "15    nonsynonymous SNV   17p13.1                .        8.327e-06   \n",
       "\n",
       "          Mut_ID count  \\\n",
       "0              .     .   \n",
       "1              .     .   \n",
       "2   COSV52800825   128   \n",
       "3   COSV52800825   128   \n",
       "4   COSV99450816     4   \n",
       "5   COSV99450816     4   \n",
       "6              .     .   \n",
       "7   COSV54240784    10   \n",
       "8   COSV50276845     6   \n",
       "9              .     .   \n",
       "10  COSV62324745    12   \n",
       "11  COSV62324745    12   \n",
       "12             .     .   \n",
       "13             .     .   \n",
       "14  COSV52708494   437   \n",
       "15  COSV52689323   855   \n",
       "\n",
       "                                                                                                                                                                                                       type  \\\n",
       "0                                                                                                                                                                                                         .   \n",
       "1                                                                                                                                                                                                         .   \n",
       "2   4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...   \n",
       "3   4x(adenocarcinoma@colon)+24x(alveolar@striated_muscle)+4x(carcinoma@breast)+8x(carcinoma@lung)+8x(embryonal@striated_muscle)+4x(fibroblastic@meninges)+32x(haemangioblastoma@blood_vessel)+8x(mening...   \n",
       "4                                                                                                                                                                                   4x(adenocarcinoma@lung)   \n",
       "5                                                                                                                                                                                   4x(adenocarcinoma@lung)   \n",
       "6                                                                                                                                                                                                         .   \n",
       "7                                                                                                                                                                       10x(haemangioblastoma@blood_vessel)   \n",
       "8                                                                                                       2x(ER-PR-positive_carcinoma@breast)+2x(adenocarcinoma@gallbladder)+2x(squamous_cell_carcinoma@lung)   \n",
       "9                                                                                                                                                                                                         .   \n",
       "10                                                                                             4x(atypical@meninges)+4x(ductal_carcinoma@breast)+4x(plasma_cell_myeloma@haematopoietic_and_lymphoid_tissue)   \n",
       "11                                                                                             4x(atypical@meninges)+4x(ductal_carcinoma@breast)+4x(plasma_cell_myeloma@haematopoietic_and_lymphoid_tissue)   \n",
       "12                                                                                                                                                                                                        .   \n",
       "13                                                                                                                                                                                                        .   \n",
       "14  38x(B_cell_prolymphocytic_leukaemia@haematopoietic_and_lymphoid_tissue)+38x(acute_myeloid_leukaemia@haematopoietic_and_lymphoid_tissue)+19x(acute_myeloid_leukaemia_associated_with_MDS@haematopoiet...   \n",
       "15  19x(ER-positive_carcinoma@breast)+19x(adenocarcinoma@bile_duct)+19x(adenocarcinoma@caecum)+19x(adenocarcinoma@colon)+19x(adenocarcinoma@gallbladder)+19x(adenocarcinoma@large_intestine)+57x(adenoca...   \n",
       "\n",
       "      icgc29_ID icgc29_Affected                                        CLNSIG  \n",
       "0             .               .                                             .  \n",
       "1             .               .                                             .  \n",
       "2      MU153762           2/170                        Uncertain_significance  \n",
       "3      MU153762           2/170                        Uncertain_significance  \n",
       "4   MU131464768           1/516                                             .  \n",
       "5   MU131464768           1/516                                             .  \n",
       "6             .               .                                             .  \n",
       "7             .               .                                        Benign  \n",
       "8   MU129512036           1/531                                             .  \n",
       "9             .               .                                             .  \n",
       "10   MU66190450           1/569  Conflicting_interpretations_of_pathogenicity  \n",
       "11   MU66190450           1/569  Conflicting_interpretations_of_pathogenicity  \n",
       "12            .               .                                             .  \n",
       "13            .               .                                             .  \n",
       "14    MU1681389            1/16                        Uncertain_significance  \n",
       "15      MU36255           1/241  Conflicting_interpretations_of_pathogenicity  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_anno.columns\n",
    "df_anno = df_anno.loc[:, ['Chr', 'Start', 'End', 'Ref', 'Alt', 'Func.ensGene34', 'Gene.ensGene34',\n",
    "        'ExonicFunc.ensGene34',\n",
    "       'cytoBand', 'gnomAD_exome_ALL', 'dbSNP154_AltFreq',\n",
    "       'Mut_ID', 'count', 'type',\n",
    "       'icgc29_ID', 'icgc29_Affected', 'CLNSIG']]\n",
    "df_anno"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8f95a80",
   "metadata": {},
   "source": [
    "### compute clinscore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e4854f7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
